spec/defines/unit_spec.rb

```diff
@@ -0,0 +1,164 @@
+require 'spec_helper'
+describe 'systemd::unit' do
+
+  let(:title) { 'spectest-unit' }
+
+  # ensure that the class is passive by default
+  describe 'with defaults for all parameters' do
+    content = <<-END.gsub(/^\s+\|/, '')
+      |[Unit]
+      |
+      |[Service]
+      |Type=simple
+      |
+      |[Install]
+    END
+
+    it { should compile.with_all_deps }
+    it { should have_systemd__unit_resource_count(1) }
+    it { should contain_file('spectest-unit_file').with({
+      'ensure'  => 'present',
+      'path'    => '/etc/systemd/system/spectest-unit.service',
+      'owner'   => 'root',
+      'group'   => 'root',
+      'mode'    => '0644',
+      'content' => content,
+    })}
+    it { should contain_service('spectest-unit_service').with({
+      'ensure'    => 'running',
+      'name'      => 'spectest-unit',
+      'enable'    => true,
+      'provider'  => 'systemd'
+    })}
+
+  end
+
+  describe 'with all parameters set' do
+    let(:params) do
+      {
+        'ensure'                  => 'present',
+        'systemd_path'            => '/tmp/systemd',
+        'unit_after'              => 'test',
+        'unit_before'             => 'test2',
+        'unit_description'        => 'Example unit',
+        'unit_requires'           => 'test3',
+        'environment'             => 'TEST=test',
+        'group'                   => 'testgroup',
+        'user'                    => 'testuser',
+        'workingdirectory'        => '/tmp/workdir',
+        'service_type'            => 'oneshot',
+        'service_timeoutstartsec' => '6',
```

Comment:
should not be an integer based on your change?

Reply:
Both integers and strings work since double() convers it. From your previous comment though this value could be string values as well.

Reply:
i know, it was just because you changed to integer and then I see it's a string here

Reply:
service_timeoutstartsec and service_startsec is tested more thoroughly on line 115 (different "time span" values).